### ISSUE TITLE
Support for assigning table entries from multi return functions

### ISSUE THREAD

**Issue by fendrin** [leafo/moonscript#247]
When using a function that returns multiple values those can't be stored directly into a table while constructing it.

valid code:
```moonscript
a, b = return_two_values!
{
  :a
  :b
  c: "whatever"
}
```
invalid code:
```moonscript
{
  a,b: return_two_values!
  c: "whatever"
}
```
What do you think about adding support for this syntax?

**Comment by Kingdaro** [comment]
Ambiguous. Consider this:
```moon
a = 5
{ a, b: something!, c: 'meh' }
```
If the support for that syntax is supported, is a 5, or the first return value from the function?

**Comment by kirbyfan64** [comment]
Maybe something better would be `{ (a, b): something!, c: 'meh' }`?

**Comment by Kingdaro** [comment]
I personally think this is enough either way.
```moon
obj = with { :some, :other, :properties }
  .x, .y = getPosition!
```
Or, for key/values on multiple lines,
```moon
obj = with { :properties, :same, :name }
  .a = 1
  .b = 2
  .c = 3
  .x, .y = getPosition!
```

**Comment by Idyllei** [comment]
> @kirbyfan64 
Maybe something better would be `{ (a, b): something!, c: 'meh' }`?

I like this syntax as it tends toward unified syntax (which *is* important). Quoting [s-ol](https://github.com/s-ol) from #203 :
> also it would be great to have multi-assignments **everywhere**:
```moonscript
if one, two = str\match "(.*).pattern.(.*)"
  print one, "okay", two
```

If Moonscript allows destructuring assignment in one place, Moonscript should allow it wherever possible (e.g. table creation).

**Comment by fendrin** [comment]
Yes, the syntax kirbyfan64 proposed seems fine. :+1:

**Comment by Kingdaro** [comment]
> also it would be great to have multi-assignments everywhere:
>
>     if one, two = str\match "(.*).pattern.(.*)"
>       print one, "okay", two

I had a question with this: would this check for both values to be true, or either? Because both cases pop up pretty often, and it's hard to tell which would happen at first glance.

**Comment by Idyllei** [comment]
> I had a question with this: would this check for both values to be true, or either?

I often see functions returning a status as the first value (e.g. pcall), and the result(s) as the following values. That creates an ambiguous situation, which I believe can be resolved.

I propose that we follow the syntax introduced in #151 for disambiguation; simply append a `?` to the end of every variable that is to be tested for truthiness before proceeding.
```moonscript
if success?, result = pcall(test)
```
would test `success`, and
```moonscript
if success?, result? = pcall(test)
```
would test both `success` and `result`.

**Comment by Kingdaro** [comment]
Yes! I agree with `?` being added as a whole. That would definitely solve the problem in an unambiguous manner.
